pkg/reconciler/ingress/ingress.go

```diff
@@ -324,10 +341,36 @@ func (r *Reconciler) reconcileRouteDeletion(ctx context.Context, ia *v1alpha1.In
 		return err
 	}
 
+	ingresses, err := r.ingressLister.Ingresses(ia.GetNamespace()).List(labels.SelectorFromSet(map[string]string{}))
+	if err != nil {
+		return fmt.Errorf("failed to list ingress in %q: %w", ia.GetNamespace(), err)
+	}
+
+	// If the Ingress is the last one in the namespace, remove its namespace from SMMR.
+	// In order to double check that we are reconciling proper ingress check with name.
+	if len(ingresses) == 1 && ia.GetName() == ingresses[0].Name {
+		smmr, err := r.smmrLister.ServiceMeshMemberRolls(smmrNamespace).Get(smmrName)
+		if err != nil {
+			return fmt.Errorf("failed to get SMMR in %q: %w", smmrNamespace, err)
+		}
+		for i, val := range smmr.Spec.Members {
+			if val == ia.GetNamespace() {
+				smmr.Spec.Members = append(smmr.Spec.Members[:i], smmr.Spec.Members[i+1:]...)
+				break
+			}
+		}
```

Comment:
Should we make a `removeIfPresent` function that does the inverse of `appendIfAbsent` and only do the update if something has really changed? Would make the code nicely symmetric.